azure_functions_worker/testutils_lc.py

```diff
@@ -0,0 +1,268 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+from typing import Dict
+
+import base64
+import json
+import os
+import re
+import subprocess
+import sys
+import time
+import uuid
+
+from Crypto.Cipher import AES
+from Crypto.Hash.SHA256 import SHA256Hash
+from Crypto.Util.Padding import pad
+import requests
+
+# Linux Consumption Testing Configs
+DOCKER_PATH = "DOCKER_PATH"
+DOCKER_DEFAULT_PATH = "docker"
+MESH_IMAGE_URL = "https://mcr.microsoft.com/v2/azure-functions/mesh/tags/list"
+MESH_IMAGE_REPO = "mcr.microsoft.com/azure-functions/mesh"
+DUMMY_CONT_KEY = "MDEyMzQ1Njc4OUFCQ0RFRjAxMjM0NTY3ODlBQkNERUY="
+
+
+class LinuxConsumptionWebHostController:
+    """A controller for spawning mesh Docker container and apply multiple
+    test cases on it.
+    """
+
+    _docker_cmd = os.getenv(DOCKER_PATH, DOCKER_DEFAULT_PATH)
+    _ports: Dict[str, str] = {}  # { uuid: port }
+    _mesh_images: Dict[str, str] = {}  # { host version: image tag }
+
+    def __init__(self, host_version: str, python_version: str):
+        """Initialize a new container for
+        """
+        self._uuid = str(uuid.uuid4())
+        self._host_version = host_version  # "3"
+        self._py_version = python_version  # "3.9"
+
+    @property
+    def url(self) -> str:
+        if self._uuid not in self._ports:
+            raise RuntimeError(f'Failed to assign container {self._name} since'
+                               ' it is not spawned')
+
+        return f'http://localhost:{self._ports[self._uuid]}'
+
+    def assign_container(self, env: Dict[str, str] = {}):
```

Comment:
This is great. How long do we wait after the container starts before sending the /assign request?

Reply:
It waits for 3 seconds.

Reply:
great. that should be good enough.

Reply:
Is there a way we can call an API in the container to check if it is active and responding? Such that we don't have to add an artificial wait?